fw/rbcx-coprocessor/src/Power.cpp

```diff
@@ -1,10 +1,305 @@
+            pinWrite(led3Pin, false);
+            buzzerSetState(false);
+        }
+    }
+}
+
+void powerInit() {
+    pinInit(
+        batteryVoltagePin, GPIO_MODE_ANALOG, GPIO_NOPULL, GPIO_SPEED_FREQ_LOW);
+    pinInit(batteryMiddleVoltagePin, GPIO_MODE_ANALOG, GPIO_NOPULL,
+        GPIO_SPEED_FREQ_LOW);
+
+    loadCalibration();
+
+    LL_ADC_InitTypeDef cfg;
+    LL_ADC_StructInit(&cfg);
+    cfg.DataAlignment = LL_ADC_DATA_ALIGN_RIGHT;
+    cfg.SequencersScanMode = LL_ADC_SEQ_SCAN_ENABLE;
+    if (LL_ADC_Init(auxiliaryAdc, &cfg) != SUCCESS)
+        abort();
+
+    LL_ADC_REG_InitTypeDef cfgReg;
+    LL_ADC_REG_StructInit(&cfgReg);
+    cfgReg.ContinuousMode = LL_ADC_REG_CONV_CONTINUOUS;
+    cfgReg.SequencerLength = LL_ADC_REG_SEQ_SCAN_ENABLE_2RANKS;
+    cfgReg.TriggerSource = LL_ADC_REG_TRIG_SOFTWARE;
+    if (LL_ADC_REG_Init(auxiliaryAdc, &cfgReg) != SUCCESS)
+        abort();
+
+    LL_ADC_INJ_InitTypeDef cfgInj;
+    LL_ADC_INJ_StructInit(&cfgInj);
+    cfgInj.TriggerSource = LL_ADC_INJ_TRIG_SOFTWARE;
+    cfgInj.TrigAuto = LL_ADC_INJ_TRIG_FROM_GRP_REGULAR;
+    if (LL_ADC_INJ_Init(auxiliaryAdc, &cfgInj) != SUCCESS)
+        abort();
+
+    LL_ADC_SetCommonPathInternalCh(ADC12_COMMON, LL_ADC_PATH_INTERNAL_VREFINT);
+
+    LL_ADC_Enable(auxiliaryAdc);
+    adcWait(LL_ADC_DELAY_ENABLE_CALIB_ADC_CYCLES);
+    LL_ADC_StartCalibration(auxiliaryAdc);
+    while (LL_ADC_IsCalibrationOnGoing(auxiliaryAdc))
+        ;
+
+    LL_ADC_SetChannelSamplingTimeFix(
+        auxiliaryAdc, batteryVoltageAdcChannel, LL_ADC_SAMPLINGTIME_1CYCLE_5);
+    LL_ADC_SetChannelSamplingTimeFix(auxiliaryAdc,
+        batteryMiddleVoltageAdcChannel, LL_ADC_SAMPLINGTIME_1CYCLE_5);
+    LL_ADC_SetChannelSamplingTimeFix(
+        auxiliaryAdc, LL_ADC_CHANNEL_VREFINT, LL_ADC_SAMPLINGTIME_71CYCLES_5);
+    LL_ADC_SetChannelSamplingTimeFix(auxiliaryAdc, LL_ADC_CHANNEL_TEMPSENSOR,
+        LL_ADC_SAMPLINGTIME_239CYCLES_5);
+
+    auxiliaryAdc->JSQR = ADC_JSQR_JL
+        | (batteryVoltageAdcChannel & ADC_JSQR_JSQ1)
+            << (ADC_JSQR_JSQ2_Pos * ((batteryVoltageAdcRank - 1) & 3))
+        | (LL_ADC_CHANNEL_VREFINT & ADC_JSQR_JSQ1)
+            << (ADC_JSQR_JSQ2_Pos * ((internalReferenceVoltageAdcRank - 1) & 3))
+        | (batteryMiddleVoltageAdcChannel & ADC_JSQR_JSQ1)
+            << (ADC_JSQR_JSQ2_Pos * ((batteryMiddleVoltageAdcRank - 1) & 3))
+        | (LL_ADC_CHANNEL_TEMPSENSOR & ADC_JSQR_JSQ1)
+            << (ADC_JSQR_JSQ2_Pos * ((temperatureAdcRank - 1) & 3));
+
+    LL_ADC_EnableIT_JEOS(auxiliaryAdc);
+    HAL_NVIC_SetPriority(
+        auxiliaryAndMotorAdcIRQn, auxiliaryAndMotorAdcIrqPrio, 0);
+    HAL_NVIC_EnableIRQ(auxiliaryAndMotorAdcIRQn);
+
+    LL_ADC_REG_StartConversionSWStart(auxiliaryAdc);
+    LL_ADC_INJ_StartConversionSWStart(auxiliaryAdc);
+
+    dispatchStatTimer.restart(dispatchStatPeriodMs);
+
+    testTask.start("test", 2, []() {
+        while (true) {
+            DEBUG("Battery: %dmv Vref: %dmv int %dmv Temp: %d | %d\n",
```

Comment:
```suggestion
            DEBUG("Battery:%dmV Vref:%dmV int:%dmV Temp:%d | %d\n",
```

Moment, tohle už tam jednou je...

Reply:
To je jen testovací loop, smažu to až to bude ready na merge.